test/client_and_receiver_integration_test.exs

```diff
@@ -164,13 +169,29 @@ defmodule SenderAndReceiverIntegrationTest do
       {:ok, %{pid: receiver_pid}} =
         MLLP.Receiver.start(port: port, dispatcher: MLLP.EchoDispatcher)
 
-      {:ok, sender_pid} = MLLP.Sender.start_link({127, 0, 0, 1}, port)
+      {:ok, sender_pid} = MLLP.Client.start_link({127, 0, 0, 1}, port)
+
+      assert MLLP.Client.is_connected?(sender_pid)
 
       MLLP.Receiver.stop(port)
       assert Process.alive?(receiver_pid) == false
 
-      assert {:error, %{type: :recv_failure, reason: :closed}} ==
-               MLLP.Sender.send_raw_and_receive_reply(sender_pid, "Simple message")
+      # We have to set up an assertion like below because we 
+      # may error out on send, but more than likely on recv.
+      # Even if we set send_timeout to 1ms it may not fail on send. It's
+      # highly depenent on the OS's gen_tcp backend, tcp/ip stack and tcp/ip
+      # settings. 
+      res =
+        case MLLP.Client.send(sender_pid, "Simple message") do
```

Comment:
Can we use this code instead. It will match on any context value unless matching on any context value is an issue

```
assert match?(
               {:error, %Error{context: _, message: "connection closed"}},
               MLLP.Client.send(sender_pid, "Simple message")
             )
```